examples/log-injection/index.js

```diff
@@ -0,0 +1,30 @@
+const { trace, SpanStatusCode, context } = require('@opentelemetry/api');
+
+const log = require('pino')();
+const doWork = () => {
+	log.info('before work');
+	while (Date.now() - start < 500) {}
+	log.info('after work work');
+};
+
+// There is no active span right now so no trace id to report.
+// Following log event will not have trace data injected.
+log.info('starting...');
+
+const start = Date.now();
+const tracer = trace.getTracer('splunk-otel-example-log-injection');
+const span = tracer.startSpan('main');
+const spanContext = trace.setSpan(context.active(), span);
+
+// This will run a function inside a context which has an active span.
+context.with(spanContext, doWork);
+
+// Even though the span has not ended yet it's not active in current context
+// anymore and thus will not be logged.
+log.info('done!');
+span.end();
+
+setTimeout(() => {
+	// wait for the spans to be flushed
```

Comment:
what about ForceFlush? [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#forceflush) [implementation](https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/src/BasicTracerProvider.ts#L152)

Reply:
Our SDK currently doesn't provide access to the NodeTracerProvider we initialize.
#331

Reply:
What about using API to get the global one?

Reply:
That currently gives us ProxyTracerProvider, which doesn't have forceFlush either and has NTP under private property. I don't have a full understanding of the correct way to solve this situation and that's why I'd like to leave it for now and figure it out after the next release.